File: 1_Review/julia/2_collection_literals.jl
# 2) Collection literals
# Run: julia 2_collection_literals.jl

# Vector (like a Python list)
# - mutable
# - can hold different types (but it's best to keep them the same)
fruit_vector = ["apple", "banana", "cherry"]

# Tuple
# - immutable (can't change its size)
fruit_tuple = ("apple", "banana", "cherry")

# Dict (like a Python dictionary)
fruit_dict = Dict("a" => "apple", "b" => "banana")

# Set (unique values)
fruit_set = Set(["apple", "banana", "cherry", "cherry"])

# Converting a tuple to a vector
fruits2 = collect(fruit_tuple)

println("***** Vector *****")
for fruit in fruit_vector
    println(fruit)
end

println("***** Tuple *****")
for fruit in fruit_tuple
    println(fruit)
end

println("***** Set *****")
for fruit in fruit_set
    println(fruit)
end

println("***** Dict *****")
# Dict iteration yields pairs: (key => value)
for (key, value) in fruit_dict
    println(key, " ", value)
end

println("Keys:")
for key in keys(fruit_dict)
    println(key)
end

println("Values:")
for value in values(fruit_dict)
    println(value)
end

File: 1_Review/javascript/2_collection_literals.js
// 2) Collection literals
// Run: node 2_collection_literals.js

// Array (like a Python list)
const fruitArray = ['apple', 'banana', 'cherry'];

// Object (key/value pairs, like a dictionary)
const fruitObject = { a: 'apple', b: 'banana' };

// Map (also key/value pairs, but supports any key type)
const fruitMap = new Map([
  ['a', 'apple'],
  ['b', 'banana'],
]);

// Set (unique values)
const fruitSet = new Set(['apple', 'banana', 'cherry', 'cherry']);

console.log('***** Array *****');
for (const fruit of fruitArray) {
  console.log(fruit);
}

console.log('***** Set *****');
for (const fruit of fruitSet) {
  console.log(fruit);
}

console.log('***** Object *****');
for (const [key, value] of Object.entries(fruitObject)) {
  console.log(key, value);
}

console.log('***** Map *****');
for (const [key, value] of fruitMap) {
  console.log(key, value);
}

console.log('Keys (Object):');
for (const key of Object.keys(fruitObject)) {
  console.log(key);
}

console.log('Values (Object):');
for (const value of Object.values(fruitObject)) {
  console.log(value);
}
